diff --git a/Readability-readerable.js b/Readability-readerable.js
index 892169fb94999987e895846b7605f9c4cf32802c..6c75e61f3c5e0b6143860fa1aa8c652548a78760 100644
--- a/Readability-readerable.js
+++ b/Readability-readerable.js
@@ -29,7 +29,7 @@ var REGEXPS = {
 function isNodeVisible(node) {
   // Have to null-check node.style and node.className.indexOf to deal with SVG and MathML nodes.
   return (!node.style || node.style.display != "none")
-    && !node.hasAttribute("hidden")
+    && !(node.hasAttribute("hidden") && !node.id.startsWith("S:"))
     //check for "fallback-image" so that wikimedia math images are displayed
     && (!node.hasAttribute("aria-hidden") || node.getAttribute("aria-hidden") != "true" || (node.className && node.className.indexOf && node.className.indexOf("fallback-image") !== -1));
 }
