diff --git a/src/index.tsx b/src/index.tsx
index 3c4a4c987d2a383aba0d9ed85fe3ca1a427c6c39..c6079fd52a038a7bf97720f4ff1d3dcae80a111a 100644
--- a/src/index.tsx
+++ b/src/index.tsx
@@ -43,6 +43,9 @@ function useVirtualizerBase<
   instance.setOptions(resolvedOptions)
 
   React.useEffect(() => {
+    if (!instance.scrollElement) {
+      instance._willUpdate()
+    }
     return instance._didMount()
   }, [])
 
