diff --git a/dist/esm/index.js b/dist/esm/index.js
index 13b1ae504e8b55efcd9d8cecd30df8ab234edc18..a8549c560b0d6b7a65b4dc58e8fb134d04dda037 100644
--- a/dist/esm/index.js
+++ b/dist/esm/index.js
@@ -22,6 +22,9 @@ function useVirtualizerBase(options) {
   );
   instance.setOptions(resolvedOptions);
   React.useEffect(() => {
+    if (!instance.scrollElement) {
+      instance._willUpdate()
+    }
     return instance._didMount();
   }, []);
   useIsomorphicLayoutEffect(() => {
