diff --git a/src/index.js b/src/index.js
index 18ee99048c31ab0e8e82601cb8021bd54cb5d780..4ff72db4c66ce9549dce240f19eef15b43a26774 100644
--- a/src/index.js
+++ b/src/index.js
@@ -125,23 +125,27 @@ const mainFunction = ({ addBase, addComponents, config }) => {
   }
 }
 
+
 module.exports = tailwindPlugin(mainFunction, {
   theme: {
     extend: {
       colors: {
         ...colorObject,
-        // adding all Tailwind `neutral` shades here so they don't get overridden by daisyUI `neutral` color
-        "neutral-50": "#fafafa",
-        "neutral-100": "#f5f5f5",
-        "neutral-200": "#e5e5e5",
-        "neutral-300": "#d4d4d4",
-        "neutral-400": "#a3a3a3",
-        "neutral-500": "#737373",
-        "neutral-600": "#525252",
-        "neutral-700": "#404040",
-        "neutral-800": "#262626",
-        "neutral-900": "#171717",
-        "neutral-950": "#0a0a0a",
+        "neutral": {
+          DEFAULT: "var(--fallback-n,oklch(var(--n)/<alpha-value>))",
+          50: "#fafafa",  
+          100: "#f5f5f5",
+          200: "#e5e5e5",
+          300: "#d4d4d4",
+          400: "#a3a3a3",
+          500: "#737373",
+          600: "#525252",
+          700: "#404040",
+          800: "#262626",
+          900: "#171717",
+          950: "#0a0a0a",
+        },
+        "neutral-content": "var(--fallback-nc,oklch(var(--nc)/<alpha-value>))",
       },
       ...utilityClasses,
     },
