diff --git a/lib/util/customConfig.js b/lib/util/customConfig.js
index 1ab15994a1e236b11fe8f3fd03404cea32c7099c..8bee7957027129f2ed555b0d128fd51f85aa0a64 100644
--- a/lib/util/customConfig.js
+++ b/lib/util/customConfig.js
@@ -14,8 +14,8 @@ try {
 const CHECK_REFRESH_RATE = 1_000;
 let previousConfig = null;
 let lastCheck = null;
-let mergedConfig = null;
-let lastModifiedDate = null;
+let mergedConfig = new Map();
+let lastModifiedDate = new Map();
 
 /**
  * @see https://stackoverflow.com/questions/9210542/node-js-require-cache-possible-to-invalidate
@@ -48,9 +48,9 @@ function loadConfig(config) {
       if (stats === null) {
         // Default to no config
         loadedConfig = {};
-      } else if (lastModifiedDate !== mtime) {
+      } else if (lastModifiedDate.get(resolvedPath) !== mtime) {
         // Load the config based on path
-        lastModifiedDate = mtime;
+        lastModifiedDate.set(resolvedPath, mtime);
         loadedConfig = requireUncached(resolvedPath);
       } else {
         // Unchanged config
@@ -91,10 +91,10 @@ function resolve(twConfig) {
     const userConfig = loadConfig(twConfig);
     // userConfig is null when config file was not modified
     if (userConfig !== null) {
-      mergedConfig = resolveConfig(userConfig);
+      mergedConfig.set(twConfig, resolveConfig(userConfig));
     }
   }
-  return mergedConfig;
+  return mergedConfig.get(twConfig);
 }
 
 module.exports = {
