diff --git a/dist/types/client/types.d.ts b/dist/types/client/types.d.ts
index b0853100a2184fa9e0e2ef5033c0adb6591b423a..4d407f5d364516b008c7afb9b862353495cbef49 100644
--- a/dist/types/client/types.d.ts
+++ b/dist/types/client/types.d.ts
@@ -43,7 +43,7 @@ type ClientResponseOfEndpoint<T extends Endpoint = Endpoint> = T extends {
     output: infer O;
     outputFormat: infer F;
     status: infer S;
-} ? ClientResponse<O, S extends number ? S : never, F extends ResponseFormat ? F : never> : never;
+} ? O : never;
 export interface ClientResponse<T, U extends number = StatusCode, F extends ResponseFormat = ResponseFormat> extends globalThis.Response {
     readonly body: ReadableStream | null;
     readonly bodyUsed: boolean;
