diff --git a/dist/immer.d.ts b/dist/immer.d.ts
index 2cb6cddda28d1fb226fc9dc5fc4b64ec7fc55a9c..161cf402a794712dc700c6b63c138622b0a3dc4c 100644
--- a/dist/immer.d.ts
+++ b/dist/immer.d.ts
@@ -48,7 +48,7 @@ type PatchListener = (patches: Patch[], inversePatches: Patch[]) => void;
  * Utility types
  */
 type PatchesTuple<T> = readonly [T, Patch[], Patch[]];
-type ValidRecipeReturnType<State> = State | void | undefined | (State extends undefined ? typeof NOTHING : never);
+export type ValidRecipeReturnType<State> = State | void | undefined | (State extends undefined ? typeof NOTHING : never);
 type ReturnTypeWithPatchesIfNeeded<State, UsePatches extends boolean> = UsePatches extends true ? PatchesTuple<State> : State;
 /**
  * Core Producer inference
