diff --git a/dist/index.js b/dist/index.js
index 147595f4a60b83ffa7316ed2c99be688e95f9fbd..74da4b11aba6a5b2d5e12bbfcf55da56015767ba 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -17,6 +17,10 @@ import {
 import { readFileSync } from "node:fs";
 import { dirname, resolve } from "node:path";
 import { fileURLToPath } from "node:url";
+
+import { createRequire } from 'node:module';
+const require = createRequire(import.meta.url);
+
 function createContext(userOptions) {
   const _dirname2 = typeof __dirname !== "undefined" ? __dirname : dirname(fileURLToPath(import.meta.url));
   const { version } = JSON.parse(
@@ -276,8 +280,8 @@ import.meta.hot.on('${DEV_PWA_ASSETS_NAME}', ({ themeColor, links }) => {
       else link.removeAttribute('type');
       if (!link.parentNode) document.head.appendChild(link);
     });
-  }  
-});  
+  }
+});
 function registerDevSW() {
   try {
     import.meta.hot.send('${DEV_READY_NAME}');
@@ -303,7 +307,7 @@ async function loadWorkboxBuild() {
     const workbox = await import("workbox-build");
     return workbox.default ?? workbox;
   } catch (_) {
-    return __require("workbox-build");
+    return require("workbox-build");
   }
 }
 async function generateRegisterSW2(options, mode, source = "register") {
